docs/CPS/CPS-XXXX/README.md

```diff
@@ -0,0 +1,129 @@
+- Funds under the control of unknown [Franken Address](https://www.essentialcardano.io/glossary/franken-address) go undetected, despite the wallet having part ownership.
+
+## Use cases
+
+### Lace Extension
+
+#### What is it doing?
+- Transaction data in responses are **transmitted as JSON**, via a **custom schema**
+- Data is sourced from the combination of [Cardano DB Sync], Ouroboros Local State Queries, and custom Ouroboros Chain-Sync projections into the PostgreSQL DB.
+- **Polling the tip for changes**. An HTTP GET request is made to a webserver every 5 seconds, checking if there's a new tip in the [Cardano DB Sync] database. If so, requests are then made to fetch transactions that are associated with the provided list of addresses after a certain point.
+- As there is **no rollback support** in the API, rather than just fetching new data, the last known point's data must be fetched again, so it can be compared with the local state to infer if a rollback has likely occurred. If it has, the state associated with the now void blocks is removed from the stores, then a new request made with the new local tip. This process repeats until an intersection is found with the server and syncing continues as per normal. The wallet ends up requesting the same data repeatedly, which is inefficient and places a higher demand on the DB with unique queries that cannot be practically cached in memory.
+
+
+> [!NOTE]
+> Requests are also made to fetch data on-demand but the scope of this document is to focus on the fundamental wallet requirements, so it's not mentioned here.
+
+<img alt="Lace fundamental state sequence flow diagram" src="images/Lace_Fundamental_State_Sequence_Flow.png">
+
+#### Why?
+- The philosophy in Lace is to place the responsibility of deriving wallet state from the historical transactions, similar to projecting state via the chain-sync protocol.
+- Blockfrost was used during the early development of the Lace frontend, however the design of the API to require many small requests conflicts with the rollback handling strategy of Lace, and therefore was not a good fit for fetching transactions. 
+- A custom API was implemented for this data access, with a batch interface, to query the `cardano-db-sync` database, and later custom projections with more a more optimised schema.
+- The API schema was implemented from scratch, based on the domain model, due to the lack of an existing standard.
+
+> [!NOTE]
+> Since this API was developed, [Cardano DB Sync] has since added a mode to store transactions in CBOR, but at the time of implementation it was not an option.
+
+#### Current Alternatives
+[Blockfrost WebSocket Link] is a partial solution for the impedance mismatch problem, however it's not addressing the fragility and standards problem. It's not practical to run this service to satisfy a single-tenant, full node, use case. 
+
+## Goals
```

Comment:
One of our goals (or an open question?) can be supporting clients who won't have a constant live connection in the background. 

I am explicitly talking about mobile devices and push messages/notifications.

Reply:
https://github.com/will-break-it/wallet-architecture/pull/3/commits/916e2390434a9fed9f51b0def55039ef3e8f4ae1